internal/pkg/proto/proto.go

```diff
@@ -48,6 +49,28 @@ func CloneAPBinding(binding *ap_binding_proto.APBinding) (*ap_binding_proto.APBi
 	}
 }
 
+func CloneFederation(federation *federation_proto.Federation) (*federation_proto.Federation, error) {
+	if clone, ok := proto.Clone(federation).(*federation_proto.Federation); !ok {
+		return nil, fmt.Errorf("bug: type assertion failed for federation %s-%s", federation.From, federation.To)
+	} else {
+		if clone == federation {
+			return nil, fmt.Errorf("bug: federation %s-%s clones are the same", federation.To, federation.To)
```

Comment:
Should this be `fmt.Errorf("bug: federation %s-%s clones are the same", federation.From, federation.To)` perhaps?

Reply:
```suggestion
			return nil, fmt.Errorf("bug: federation %s-%s clones are the same", federation.From, federation.To)
```